semgrep-core/src/engine/Match_tainting_rules.ml

```diff
@@ -98,9 +98,10 @@ let any_in_ranges rule any rwms =
    * the AST at some point. *)
   match Visitor_AST.range_of_any_opt any with
   | None ->
-      logger#debug
-        "Cannot compute range, there are no real tokens in this AST: %s"
-        (G.show_any any);
+      if any <> G.Anys [] then
```

Comment:
Good question. I added a clarification comment. This is because I convert `orig` into `any` just for the purpose of checking for taint labels, and `NoOrig` becomes `Any []`, so there are a lot of those pointless warnings.